packages/smithy-client/src/date-utils.spec.ts

```diff
@@ -36,6 +41,80 @@ describe("parseRfc3339DateTime", () => {
     "1985-04-05T23:61:50.52Z",
     "1985-04-05T23:20:61.52Z",
     "1985-04-31T23:20:50.52Z",
+    "1996-12-19T16:39:57",
+    "2019-12-16T22:48:18+02:04",
+    "2019-12-16T22:48:18-01:02",
```

Comment:
Add a `// this is allowed by parseRfc3339DateTimeWithOffset but rejected by parseRfc3339DateTime` to make these stand out.
Moreover, instead of adding these 2 cases in the middle of all the cases that are repeated for `parseRfc3339DateTimeWithOffset`, keep them at the end, so the cases that are common are more easily maintainable (e.g., if a new test case needs to be added to both later, it will be easy to copy-paste). Or even share an array variable in code.

Reply:
I went ahead moved the common invalid cases to a shared array.